pallets/asset/src/lib.rs

```diff
@@ -1187,10 +1198,33 @@ impl<T: Trait> AcceptTransfer for Module<T> {
     fn accept_asset_ownership_transfer(to_did: IdentityId, auth_id: u64) -> DispatchResult {
         Self::_accept_token_ownership_transfer(to_did, auth_id)
     }
+
+    fn update_balance_of_scope_id(
+        of: ScopeId,
+        target_did: IdentityId,
+        ticker: Ticker,
+    ) -> DispatchResult {
+        let current_balance = Self::balance_of(ticker, target_did);
+        let balance_at_scope = Self::balance_of_at_scope(of, target_did);
+        // Conditions to check.
+        // 1. Check of `current_balance` is used to skip storage changes when user has 0 current balance.
+        // 2. Used `balance_at_scope` variable to skip re-updating the aggregate balance of the given identityId whom
+        // has the scope claim already.
```

Comment:
I am failing to understand these conditions
1) Why do we skip storage write even if the balance is 0?
2) Why will an identity have a balance at scope without a scope claim?

Reply:
1. When a new investor is providing a scope_claim to itself, It means it will have a 0 balance of the give scopeId or ticker.
2. There is a chance that users remove its scope claim (This will grief themselves as they aren't able to transact given asset anymore) & re-added it in that case we don't want the re-update its balances as storage is already up to date.

Reply:
1. Yeah but why do we skip the variable update? Don't we want the UI to know that this identity also belongs to the scopeid?
2. I don't think we should be worrying about this case. It's much cleaner to just update the variable. There's no hard done even if we update the variable in the vase you described, right? (apart from a storage write).

Reply:
1. I agree for the case 1 we can update it for the 0 balance.
2. We don't want it because of this - 

```
<AggregateBalance<T>>::mutate(ticker, of, |bal| *bal = *bal + current_balance);
```

Above is mutating the aggregate balance so if I keep doing claim addition and removal continuously then this will increase continously with non-existent balance.